docs/release-checklist.md

```diff
@@ -0,0 +1,36 @@
+    1. Might need to run a `dotnet refresh` to update the packages.lock.json files
+    1. Might also need to update baseline tests (run `bicep/scripts/SetBaseline.ps1`)
+1. Verify the latest build on the `main` branch is green: ![Build on main](https://github.com/Azure/bicep/actions/workflows/build.yml/badge.svg?branch=main).
+1. Review history for changes to [bicepconfig.schema.json](https://github.com/Azure/bicep/commits/main/src/vscode-bicep/schemas/bicepconfig.schema.json). Raise an issue for any recently-added linter rules which do not have public documentation.
+1. Bump the version number by incrementing the minor version number in [this file](https://github.com/Azure/bicep/blob/main/version.json) (example [here](https://github.com/Azure/bicep/pull/9698)) (**only required for end-of-month releases**)
+1. Run the Official Build (see [this README](https://msazure.visualstudio.com/One/_git/BicepMirror) for instructions).
+1. Get version number from official build and push a new tag to the Bicep repo. This should be of format `vXX.YY.ZZ` - e.g `v0.14.85`.
+1. [Create a draft release](https://github.com/Azure/bicep/releases/new) for the new tag and set release title to the tag name. Use the "Save draft" button to save the changes without publishing it.
+1. Run `bicep/scripts/CreateReleaseNotes -FromTag <previous tag> -ToTag <new tag>` and set the output as the release description.
+    1. Give the output of this script to a PM, and ask them to clean up the notes for the release.
+    1. Once they have cleaned up the notes, copy + paste them into the draft notes, and hit "Save draft" again.
+1. Run `BicepMirror/scripts/UploadSignedReleaseArtifacts.ps1` to add official artifacts to the release.
+    * `-WorkingDir` can be any empty temporary directory
+    * `-BuildId` is only needed if the latest official build is NOT the official build you are trying to release
+1. Validate VSCode extension and Bicep CLI manually on Windows, Mac & Linux:
+    1. Download `vscode-bicep.vsix` from the draft release, and [Install it from VSIX](https://code.visualstudio.com/docs/editor/extension-marketplace#_install-from-a-vsix). Verify that you can open a Bicep file, that text is correctly colorized, and that error messages show up as expected.
+    1. Download the appropriate Bicep executable for your platform (e.g. `bicep-linux-x64`). Verify you can invoke it with e.g. `bicep-linux-x64 --version`, and that it prints the expected output.
+1. Publish the release on GitHub.
+1. Upload copyleft dependency source to 3rd party disclosure site. See [instructions](https://msazure.visualstudio.com/One/_wiki/wikis/Azure%20Deployments%20Team%20Wiki/369910/Bicep-release-step-Upload-copyleft-source-to-3rd-party-disclosure-site).
+1. Upload vscode-bicep.VSIX to the VS marketplace [here](https://marketplace.visualstudio.com/manage).
+1. Upload vs-bicep.VSIX to VS marketplace
```

Comment:
@bhsupra Can you comment on the differences in VS VSIX upload from the VS Code one, so @anthony-c-martin can update the step. We talked about the steps being slightly different on Teams yesterday.

Reply:
@bhsubra FYI. I've set the PR to auto-merge but am happy to send out a new one if you're able to give more detail.